netket/models/equivariant.py

```diff
@@ -33,19 +33,14 @@
     DenseSymmFFT,
     DenseEquivariantFFT,
     DenseEquivariantIrrep,
+    default_gcnn_initializer,
```

Comment:
Ah, I just noticed now, but...
in principle all things you are importing here from `symmetric_linear` are 'implementation details. You should only access `netket.nn.DenseSymm` and `netket.nn.DenseEquivariant`.

While this is intra-netket and using internals is fine, I'd like to avoid it if possible.

For what regards this particular PR, I'd be fine with not importing the default initialiser and just redefining it here + a comment below.

Then, it would be great if in a follow up PR you could just use the standard API `DenseSymm(mode="FFT")` or `DenseEquivariant(mode=irrep)` instead of using the type directly.

This (1) tests the public API, so it's good, (2) makes us use our public API and (3) disentangles the internals of the two modules.

Moreover, maybe by doing this we can get rid of having 4 different GCNN classes?

Reply:
`DenseSymm()` and `DenseEquivariant()` take things that don't play nicely at all with jitting, so if you put these functions in `setup()` of another module, their construction gets very slow. I seem to recall we tried this at some point (but @chrisrothUT  might remember better) and decided against it when we saw the performance.